src/frontend/src/session.rs

```diff
@@ -455,8 +451,7 @@ pub struct SessionImpl {
 
     /// Shutdown channels map
     /// FIXME: Use weak key hash map to remove query id if query ends.
-    shutdown_receivers_map:
-        Arc<tokio::sync::Mutex<HashMap<QueryId, Option<mpsc::Sender<QueryMessage>>>>>,
+    query_executions_map: Arc<tokio::sync::Mutex<HashMap<QueryId, Arc<QueryExecution>>>>,
```

Comment:
Have tried WeakValueHashMap (when QueryExecution ends, remove entry lazily). But can not achieve RAII : The QueryExecution life cycle is not long enough.

Reply:
This map should not exists in `SessionImpl` but `QueryManager`